services/workers/src/main/java/org/sagebionetworks/file/worker/PreviewWorker.java

```diff
@@ -91,10 +93,13 @@ public void run(ProgressCallback progressCallback, ChangeMessage changeMessage)
 					true);
 			throw new RecoverableMessageException();
 		} catch (Throwable e) {
-			// If we do not know what went wrong then we do no re-try
-			log.error("Failed to process message: " + changeMessage.toString(), e);
-			workerLogger.logWorkerFailure(this.getClass(), changeMessage, e,
-					false);
+			if (e.getCause() instanceof IIOException) {
```

Comment:
From http://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.20.2, no need to check for null, instanceof will be false in that case.